### PR TITLE
lesson_4

### PR DESCRIPTION
**Pull request by Reallocman**

### CODE REVIEW

File: Task_4/main.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+type Shape interface {
+	// add appropriate methods
```
Comment (rat6666) [inline Task_4/main.go]: meaningless comment

File: Task_4/main.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	height float64
+	width  float64
+}
+func DescribeShape(s Shape) {
+	fmt.Println(s)
+	fmt.Printf("Area: %.2f\n", s.Area())
+	fmt.Printf("Perimeter: %.2f\n", s.Perimeter())
+}
+func main() {
+	// choose your own dimensions
```
Comment (rat6666) [inline Task_4/main.go]: meaningless comment

File: Task_4/main.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+}
+func(r Rectangle) Perimeter() float64{
+	return 2*(r.width+r.height)
+}
+func (c Circle) String() string{
+	return fmt.Sprintf("Circle:radius %.2f",c.radius)
+}
+func (r Rectangle) String() string{
+return fmt.Sprintf("Rectangle with height %.2f and width %.2f",r.height,r.width)
+}
```
Comment (rat6666) [inline Task_4/main.go]: Why is everything in one file? This was just an example. Please, move interface and entities in different files/packages

**Review by rat6666 — CHANGES_REQUESTED** [review]

**Review by rat6666 — APPROVED** [review]

File: Task_4/shapes/shapes.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+func DescribeShape(s Shape) {
+	fmt.Println(s)
+	fmt.Printf("Area: %.2f\n", s.Area())
+	fmt.Printf("Perimeter: %.2f\n", s.Perimeter())
+}
+
+
+
```
Comment (dimonl) [inline Task_4/shapes/shapes.go]: remove empty lines in the end of the file

**Review by dimonl — APPROVED** [review]

**Pull request merged**